### ISSUE TITLE
`no-descending-specificity` is reporting false positive

### ISSUE THREAD

**Issue by vilchik-elena** [stylelint/stylelint#5676]
There are false positives for `no-descending-specificity` rule for the following case
```css
div p > a { }
div > a {  }
```
Here `a` is child of `p` or `div` thus those two rules can never apply at the same time

Config
```json
{
  "rules": {
    "no-descending-specificity": true
  }
}
```

> Which version of Stylelint are you using?
14.0.1 (using Demo)
https://stylelint.io/demo/#N4Igxg9gJgpiBcIoEsBuACADugfOghusOgL4A6AdpShnocaZSADQgDOAnhQC74AeCcGzYtwECgDNkAc0HBK6dGRAAnAK4AbGG2XwiCxUpAUIAWlhswMCigrTTbTDDDIpL7h13pu6mAfIUJCAkQA

The following warnings were flagged

```shell
3:1
error
Expected selector "div > a" to come before selector "div p > a" (no-descending-specificity)
```

**Comment by thw0rted** [comment]
Hi @jeddy3 , I originally reported this error to Elena's team [here](https://community.sonarsource.com/t/css-ordering-rule-incorrectly-flags-selectors-that-dont-match/30989) -- SonarQube is using stylelint, apparently.

It looks like my problem is more or less an exact duplicate of #3072.   Your team also closed that one, because the rule "checks specificity only", but as the author pointed out, it would be a lot more useful if it could ignore pairs of rules that can be statically determined (*without* inspecting HTML/DOM) as impossible to apply to the same element, like `some-tag > a {} other-tag > a {}`.

There's some confusion because in both the linked issues, my problem is compared to `body .x > a {} .y > a {}`, but it's a totally different class of problem.  The latter example *should* be flagged, because an element can have as many classes as you like, which means you would be surprised by the rule that "wins" for `<div class="x y"><a></a></div>`.  This is not the case for rules where the parent side of the direct-child selector includes differing tags.

So: I consider the current behavior to be a bug, or at least a shortcoming that should be feasible to address.  It is *not* caused by the limitation described in the docs, specifically for the direct-child combinator where parent tags differ, because (unless I grossly misunderstand how CSS works) two different tag selectors can never apply to the same parent node.  Given that, I think either #3072 or this issue should be re-opened.

**Comment by vvdevteam** [comment]
still, an issue, may consider adding something like ` "no-descending-specificity": [true, { "ignorePseudoClasses": true }]`?
